.github/workflows/tests.yml

```diff
@@ -31,29 +31,43 @@ jobs:
           - ubuntu-latest
           - macos-latest
           - windows-latest
-        python-version:
-          - "3.7"
-          - "3.8"
-          - "3.9"
-          - "3.10"
-          - "3.11"
-          - "pypy-3.7"
 
     steps:
       - name: "Check out the repo"
         uses: "actions/checkout@v3"
 
       - name: "Set up Python"
+        id: "setup-python"
         uses: "actions/setup-python@v4"
         with:
-          python-version: "${{ matrix.python-version }}"
-          cache: pip
-          cache-dependency-path: 'requirements/*.txt'
+          # The last listed Python version is the default.
+          python-version: |
+            pypy-3.7
+            3.7
+            3.8
+            3.9
+            3.10
+            3.11
+
+      - name: "Restore cache"
+        id: "restore-cache"
+        uses: "actions/cache@v3"
+        with:
+          path: |
+            .tox/
+            .venv/
+          key: "cache-python-${{ steps.setup-python.outputs.python-version }}-os-${{ runner.os }}-hash-${{ hashFiles('tox.ini', 'requirements/*.txt') }}"
+
+      - name: "Identify venv path"
+        shell: "bash"
+        run: "echo 'venv-path=${{ runner.os == 'Windows' && '.venv/Scripts' || '.venv/bin' }}' >> $GITHUB_ENV"
```

Comment:
This step establishes the relative path to the Python executable and writes it to `$GITHUB_ENV`, which expects INI-like lines that establish an environment variable name and its value. Subsequent steps will have access to a `venv-path` environment variable.

Although the default shell is specified above to be `bash`, note that the forward slashes would also work in Windows' default shell, Powershell, so no changes are needed here.

However, I explicitly included the `shell: "bash"` configuration because the test suite failed for me locally with "path not found" errors while running in Powershell, so I'm anticipating the possibility that using bash as the default shell even when running on Windows is masking a failure. Investigating that was not in-scope for this work, but I tried to anticipate a future change to running Powershell on Windows in CI.